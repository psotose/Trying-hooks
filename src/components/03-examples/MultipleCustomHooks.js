import React from 'react';
import { useFetch } from '../../hooks/useFetch';
import { useCounter } from '../../hooks/useCounter';
import '../01-useState/counter.css';

export const MultipleCustomHooks = () => {

  const { counter, increment } = useCounter(1);

  const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
  
  const { author, quote } = !!data && data[0];

  return (
    <div>
      <h1>BreakingBad Quotes</h1>
      <hr />
      {
        loading
          ?
            (<div className="alert alert-info text-center"> Loading... </div> )
          :
            (
              <blockquote className="blockquote text-end">
                <p className="mb-0"> {quote} </p>
                <br/>
                <footer className="blockquote-footer"> { author } </footer>
              </blockquote>
            )
      }

      <button 
        className="btn btn-primary"
        onClick={increment}
      > 
        Next quote 
      </button>
    </div>
  )
}
