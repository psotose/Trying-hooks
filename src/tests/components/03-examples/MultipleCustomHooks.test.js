import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Testing <MultipleCustomHooks />', () => {

  useCounter.mockReturnValue({
    counter: 10,
    increment: () => {}
  });

  test('should be render correctly', () => {

    useFetch.mockReturnValue({
      data: null, 
      loading: true, 
      error: null
    });

    const wrapper = shallow( <MultipleCustomHooks /> );
    expect( wrapper ).toMatchSnapshot();
  });

  test('should show info', () => {
    useFetch.mockReturnValue({
      data: [{
        author: 'Patito',
        quote: 'Hola Mundo'
      }], 
      loading: false, 
      error: null
    });

    const wrapper = shallow( <MultipleCustomHooks /> );
    expect( wrapper.find('.alert').exists() ).toBe( false );
    expect( wrapper.find('.mb-0').text().trim() ).toBe( 'Hola Mundo' );
    expect( wrapper.find('footer').text().trim() ).toBe( 'Patito' );
    
  })
});