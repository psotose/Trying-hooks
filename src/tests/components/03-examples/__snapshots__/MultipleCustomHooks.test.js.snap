// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <MultipleCustomHooks /> should be render correctly 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
     Loading... 
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next quote
  </button>
</div>
`;
